Create/delete/describe vpc, subnet, routetable, internetgateway, volume and tags endpoints

Davinder2me opened the pull request:
New Endpoints for Aws-Mock:

CreateVpc
DeleteVpc
CreateSubnet
DeleteSubnet
CreateRouteTable
DeleteRouteTable
CreateRoute
CreateInternetGateway
DeleteInternetGateway
AttachInternetGateway
CreateVolume
DeleteVolume
CreateTags
DeleteTags
DescribeTags

maxiaohao commented on config/checkstyle/checkstyle.xml:
```diff
@@ -118,7 +118,11 @@
+        <module name="MethodLength">
+            <property name="max" value="400" />       
+        </module>
+        
```
Such empty line with trailing space should be eliminated in code - please fix here and elsewhere.

maxiaohao commented on src/integration-test/java/com/tlswe/awsmock/cloudwatch/CloudWatchBaseTest.java:
```diff
@@ -138,16 +135,16 @@ public static void setup() {
+
         Datapoint dataPoint = null;
-       
+
```
Yes like this (eliminating any trailing spaces).

maxiaohao commented on src/integration-test/java/com/tlswe/awsmock/cloudwatch/CloudWatchBaseTest.java:
```diff
@@ -159,8 +156,12 @@ protected final Datapoint getMetricStaticticsTest(String metricName) {
             dataPoint = result.getDatapoints().get(0);
+
+            for (Datapoint dp : result.getDatapoints()) {
+                System.out.println(dp.getTimestamp());
```
Such debug println may need to be commented or it can pollute the build/test log.

maxiaohao commented on src/integration-test/resources/aws-mock.integration-test.properties:
```diff
@@ -5,9 +5,11 @@ aws.accessKey=foo
 
 ec2.endpoint=http://localhost:8001/aws-mock/ec2-endpoint
+#ec2.endpoint=http://54.193.73.100/aws-mock/ec2-endpoint
 cloudwatch=http://localhost:8001/aws-mock/cloudwatch
```
This is great to run integrated-test against cloudWatch on localhost:8001!

maxiaohao commented on src/main/java/com/tlswe/awsmock/cloudwatch/servlet/MockCloudWatchEndpointServlet.java:
```diff
@@ -49,8 +47,8 @@ protected final void doGet(final HttpServletRequest request,
-        for (String key: queryParams.keySet()) {
-            System.out.println(key + " : "  + queryParams.get(key)[0]);
+        for (String key : queryParams.keySet()) {
+            System.out.println(key + " : " + queryParams.get(key)[0]);
```
Such debug println in endpoint servlet should also be removed or commented out.

maxiaohao commented on src/main/java/com/tlswe/awsmock/ec2/control/MockVpcController.java:
```diff
@@ -0,0 +1,132 @@
+        ret.setVpcId("vpc-" + UUID.randomUUID().toString().substring(0, VPC_ID_POSTFIX_LENGTH));
+        ret.setInstanceTenancy(instanceTenancy);
+        // Make sure only one VPC is default.
+        if (allMockVpcInstances.size() == 0) {
```
Minor thought here: since allMockVpcInstances.size() can be called in a non-sync way, there may be multiple MockVpcs set as default. You may verify that using intensive tests that creates vpcs and may need to make this `if` block synced.

maxiaohao commented on src/main/java/com/tlswe/awsmock/ec2/servlet/MockEc2EndpointServlet.java:
```diff
@@ -47,14 +45,17 @@ protected final void doGet(final HttpServletRequest request,
                 .getParameterMap();
 
+        for (String key : queryParams.keySet()) {
+            System.out.println(key + " : " + queryParams.get(key)[0]);
```
This unexplained debug msg by println may also need to be removed or commented.

maxiaohao replied:
[CodeEditorSettingsInEclipse.docx](https://github.com/treelogic-swe/aws-mock/files/830265/CodeEditorSettingsInEclipse.docx)

Davinder2me commented:
Thanks for review my code changes. I have updated the code with review comments. Please review.

Davinder2me commented:
Added the support for pagnation for describeVolumes. I have updated the ec2 wsdl file. Please update the maven central with latest jars cxf. Please review the changes.

Davinder2me commented:
Hi, 

I have added few more endpoints:
Add the supoort for describeralarms, createsecuritygroup, deletesecuritygroup, authorizeIngress, Egress. Added the unit and integration tests to make code coverage 80%.

Please review code as this pull request for 25 days.

Thanks
Davinder

nottoseethesun commented:
Thanks!  We'll get to it right away.

maxiaohao commented on cxf-stub/ec2-2014-02-01.wsdl:
```diff
@@ -1443,6 +1443,7 @@
       <xs:complexType name="DescribeVolumesResponseType">
         <xs:sequence>
           <xs:element name="requestId" type="xs:string"/>
+          <xs:element name="nextToken" type="xs:string" minOccurs="0"/>
```
This field has been added in recent AWS API, right?
If you can find a lately full version of new ec2-xxx.wsdl with such new fields, that would be great!

Davinder2me commented:
Yes, this is new field. However I searched lot. I could not find latest ec2.xxxx.wsdl file. If you know that location, I can update the pull request.

Thanks

maxiaohao commented:
Unfortunately I don't have the recent wsdl file either, so I am now going to update cxf jar, upload it to maven central and check the build.

maxiaohao commented:
Oops, the oss.sonatype.org control panel isn't reflecting my updated staging cxf-stub.jar uploaded for now. I'd try later. 
Btw, I think as sub project 'cxf-stub' is used by aws-mock only, why not build cxf-stub.jar each time we build parent project aws-mock? (avoid downloading cxf-stub from maven repo anymore)
I know this kind of 'chained' child-parent builds are widely used in maven projects, but I haven't practiced yet with gradle. Are you interested in converting aws-mock into 'chained' triggered build for cxf-stub on each build?

Davinder2me commented:
Yes , I am ok with chained build.
Thanks

maxiaohao commented:
Thanks Davinder2me for your continuous contribution to aws-mock!
The coverage level is over 90%, which is amazing! And I opened issue #44 for making the "chained" child-parent builds, you may contribute to that if you are interested in it:)
Thanks again for your great help!
